### PR TITLE
Migrate the Antenna reporter to the new license API

### PR DESCRIPTION
**Pull request by mnonnenmacher**
See the commit messages for details.

### CODE REVIEW

File: spdx-utils/src/main/kotlin/SpdxExpression.kt [hunk 1]
```diff
@@ -278,6 +278,11 @@ sealed class SpdxSingleLicenseExpression : SpdxExpression() {
         fun parse(expression: String): SpdxSingleLicenseExpression =
             SpdxExpression.parse(expression) as SpdxSingleLicenseExpression
     }
+
+    /**
+     * Return the string identifier of this license, without the optional exception.
+     */
+    abstract fun licenseId(): String
```
Comment (sschuberth) [inline spdx-utils/src/main/kotlin/SpdxExpression.kt:285]: How about calling this `simpleLicense` to resemble `SpdxSimpleExpression`?

File: spdx-utils/src/main/kotlin/SpdxExpression.kt [hunk 2]
```diff
@@ -278,6 +278,11 @@ sealed class SpdxSingleLicenseExpression : SpdxExpression() {
         fun parse(expression: String): SpdxSingleLicenseExpression =
             SpdxExpression.parse(expression) as SpdxSingleLicenseExpression
     }
+
+    /**
+     * Return the string identifier of this license, without the optional exception.
```
Comment (sschuberth) [inline spdx-utils/src/main/kotlin/SpdxExpression.kt:283]: How about just saying "without any license exception" (and no comma before that)?

File: spdx-utils/src/main/kotlin/SpdxExpression.kt (hunk 1 above)
Comment (mnonnenmacher) [reply spdx-utils/src/main/kotlin/SpdxExpression.kt]: Done.

File: spdx-utils/src/main/kotlin/SpdxExpression.kt (hunk 2 above)
Comment (mnonnenmacher) [reply spdx-utils/src/main/kotlin/SpdxExpression.kt]: Done.

**Pull request merged**